src/pool/pool_jemalloc.c

```diff
@@ -461,7 +467,19 @@ static umf_result_t op_initialize(umf_memory_provider_handle_t provider,
         err = je_mallctl("arenas.create", (void *)&arena_index, &unsigned_size,
                          NULL, 0);
         if (err) {
-            LOG_ERR("Could not create arena.");
+            // EAGAIN - means that a memory allocation failure occurred
+            // (2 * utils_get_num_cores()) is the required minimum number of arenas
+            if (err == EAGAIN && (i >= (2 * utils_get_num_cores()))) {
```

Comment:
If user set arenas number param, we should create all of them or fail.

Reply:
Done